registry/registry.go

```diff
@@ -126,16 +136,23 @@ func (c *client) GetRepository(repository string) (_ []flux.ImageDescription, er
 	// Add the backoff mechanism so we don't DOS registries
 	transport = BackoffRoundTripper(transport, initialBackoff, maxBackoff, clockwork.NewRealClock())
 
-	client := &dockerregistry.Registry{
-		URL: httphost,
-		Client: &http.Client{
-			Transport: roundtripperFunc(func(r *http.Request) (*http.Response, error) {
-				return transport.RoundTrip(r.WithContext(ctx))
-			}),
-			Jar:     jar,
-			Timeout: 10 * time.Second,
+	var client backend = herokuWrapper{
```

Comment:
maybe just a separate commit? but the motivation is coupled to serializing data to send to memcache. so, I'm not sure.